WebUI/Data/DataAccess/WishedItemService.cs

```diff
@@ -0,0 +1,49 @@
+﻿using ConestogaInsidersClub.Data.Models;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ConestogaInsidersClub.Data.DataAccess
+{
+    public class WishedItemService : IWishedItemService
+    {
+        private ApplicationDbContext context;
+        public WishedItemService(ApplicationDbContext context)
+        {
+            this.context = context;
+        }
+
+        public Task<List<Game>> GetWishedGamesForUser(string userId)
+        {
+            return context.WishedItems
+             .Where(w => w.UserId == userId)
+             .Select(w => w.Game)
+             .ToListAsync();
+        }
+
+        public async Task AddToWishlist(Game game, string userId)
+        {
+            context.WishedItems.Add(new WishedItem
+            {
+                GameId = game.GameId,
+                UserId = userId
+            });
+            await context.SaveChangesAsync();
+        }
+
+        public async Task RemoveFromWishlist(Game game, string userId)
+        {
+            var wishedItem = await context.WishedItems.FirstOrDefaultAsync(w => w.GameId == game.GameId);
+            context.WishedItems.Remove(wishedItem);
+            await context.SaveChangesAsync();
+        }
+
+        public async Task AddOrRemoveFromWishlist(Game game, string userId)
```

Comment:
What is the use case for this method?

Reply:
Well, I assumed you would have some sort of a star on the Game page so you could just click it and call this method. This way the frontend doesn't care about whether the game is already in the user's wishlist or not